pkg/metrics/gameservers_count.go

```diff
@@ -24,16 +24,23 @@ import (
 )
 
 // GameServerCount  is the count of gameserver per current state and per fleet name
-type GameServerCount map[agonesv1.GameServerState]map[string]int64
+type GameServerCount map[agonesv1.GameServerState]map[fleetKey]int64
+
+type fleetKey struct {
+	name      string
+	namespace string
+}
 
 // increment adds the count of gameservers for a given fleetName and state
-func (c GameServerCount) increment(fleetName string, state agonesv1.GameServerState) {
+func (c GameServerCount) increment(fleetName, fleetNamespace string, state agonesv1.GameServerState) {
+	key := fleetKey{name: fleetName, namespace: fleetNamespace}
```

Comment:
Cool to use a struct as a key. 👍

Reply:
Let's cover a whole gameserver_countfile with tests in a separate PR.